robot/impl/local_robot.go

```diff
@@ -1648,6 +1632,119 @@ func (r *localRobot) reconfigure(ctx context.Context, newConfig *config.Config,
 	}
 }
 
+func (r *localRobot) reconfigureTracing(ctx context.Context, newConfig *config.Config) {
+	newTracingCfg := newConfig.Tracing
+	hasDiff := newTracingCfg != r.mostRecentCfg.Load().(config.Config).Tracing
+	if !hasDiff {
+		return
+	}
+	if !newTracingCfg.IsEnabled() {
+		prevExporters := trace.ClearExporters()
+		for _, ex := range prevExporters {
+			//nolint: errcheck, gosec
+			ex.Shutdown(ctx)
+		}
+		r.traceClients.Store(nil)
+		return
+	}
+	var exporters []sdktrace.SpanExporter
+	var robotTraceClients []otlptrace.Client
+	if newTracingCfg.Disk {
+		func() {
+			partID := "local-config"
+			if newConfig.Cloud != nil {
+				partID = newConfig.Cloud.ID
+			}
+			tracesDir := filepath.Join(utils.ViamDotDir, "trace", partID)
+			if err := os.MkdirAll(tracesDir, 0o700); err != nil {
+				r.logger.Errorw("failed to create directory to store traces", "err", err)
+				return
+			}
+			r.logger.Infow("created trace storage dir", "dir", tracesDir)
```

Comment:
This feels like a diagnostic log to me. What do we think about making a `r.logger.Sublogger("traces")` and using that here and for any other trace-related logs? We could include it as diagnostic in the app list. (Error logs are, of course, always user-facing.)